soco/core.py

```diff
@@ -1490,6 +1493,59 @@ def get_item_album_art_uri(self, item):
         else:
             return None
 
+    @only_on_master
+    def set_sleep_timer(self, sleep_time_seconds):
+        """Sets the sleep timer.
+
+        Args:
+            sleep_time_seconds (`int`): How long to wait before turning off
+                speaker in seconds. Maximum value of 86399
+
+        Returns:
+            bool: True if succesful, False otherwise
+
+        Raises SoCoException (or a subclass) upon errors, ValueError for
+            incorrect syntax.
+
+        """
+        if sleep_time_seconds and \
```

Comment:
I'm wondering whether the checks and the exception throwing could be refactored slightly, to become a little cleaner.

How about:
 1. Doing only the second check here, the "can accurately be interpreted as integer" check"
 2. Then convert to int and then to '1:00:00' string
 3. Then simply make the self.avTransport.ConfigureSleepTimer call inside an try except. Assuming you get a different exception with a bad value than with e.g a communications error, the you could check for the former and raise a ValueError and reraise the latter. Let me know if I'm not making any sense and I can exemplify it.

Note, that this is just a suggestion.